datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -2921,17 +2922,53 @@ pub enum Partitioning {
     DistributeBy(Vec<Expr>),
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum ColumnUnnestType {
+    List(Vec<ColumnUnnestList>),
+    // for struct, there can only be one unnest performed on one column at a time
+    Struct,
+    // Infer the unnest type based on column schema
+    // If column is a list column, the unnest depth will be 1
+    // This value is to support sugar syntax of old api (unnest(columns1,columns2))
```

Comment:
I think we didn't support multi expression in unnest (?

Reply:
tks, it is a typo

and my implementation is hard to read right now, so reviewing it can be exausting :pray: 
I'm still refactoring/simplifying to make it more readable